projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/listeners/Function.java

```diff
@@ -982,6 +982,20 @@ private static int getAlign(long argument) {
         return (int) argument & (Long.SIZE - 1);
     }
 
+    private static Type getExtractedElementType(Type type) {
+        Type valType = type;
+        if (valType instanceof PointerType) {
```

Comment:
According to [the llvm documentation](http://releases.llvm.org/4.0.1/docs/LangRef.html#extractelement-instruction) and also looking at the llvm 4.0.1 source code this should only ever be a vector type to be valid bitcode. Did you actually see these in your bitcode files? In any case, these are all subclasses of AggregateType which declares the 'getElementType(int)' method. Could you change this to `if (valType instanceof AggregateType) val = ((AggregateType) val).getElementType(0)`.